.travis.yml

```diff
@@ -114,47 +108,65 @@ matrix:
 
   # Job 11: Run Clang-Tidy
   - os: linux
-    env: T="Clang-Tidy" TEST=tidy V=1 PATH="/usr/bin:$PATH" NO_COMPILE=1
+    compiler: clang
+    env: T="Clang-Tidy" CMAKE_PARAMS="-DENABLE_CLANG_TIDY=ON" RUN_TESTS=0
+    addons:
+      apt:
+        packages:
+        - libzstd1-dev
+        - libb2-dev
+
+  # Job 12: Run Clang-Format amd CMake-Format check mode
+  - os: linux
+    compiler: clang
+    env: T="Clang-Format" BUILDEXTRAFLAGS="--target check_format" RUN_TESTS=0
     addons:
       apt:
         packages:
-        - elfutils
         - libzstd1-dev
         - libb2-dev
     before_install:
-    - pip install --user compiledb
+    - pip install --user cmake_format
 
-  # Job 12: Run Clang-Format
+  # Job 13: Build binary package, extract it out of source, run tests.
   - os: linux
-    env: T="Clang-Format" V=1 PATH="/usr/bin:$PATH" NO_COMPILE=1 TEST=check_format
+    env: T="Build binary and verify" SPECIAL=build_and_verify_package
     addons:
       apt:
         packages:
         - elfutils
         - libzstd1-dev
         - libb2-dev
 
-  # New jobs go here until they are established for some weeks
-  allow_failures:
```

Comment:
Is it intentional to add Clang-Tidy and Clang-Format as required tests? I'm OK with that, but it feels like that should be a separate PR.